File: android/app/src/main/java/com/dailyplan/app/util/ASRSplitConfig.kt
// ASRSplitConfig.kt
// 双端共享 ASR 拆分配置解析（规格 §8）。M1 仅定义结构、F2 才消费。
// 禁止在代码中硬编码拆分标点集或阈值，必须从 assets/asr_split_config.json 解析（P0-4）。

package com.dailyplan.app.util

import android.content.Context
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.Json
import java.io.IOException

@Serializable
data class ASRSplitConfig(
    val configVersion: Int,
    val splitPunctuation: List<String>,
    val splitPauseThresholdMs: Int,
    val includeEnumerationComma: Boolean,
    val includeNewline: Boolean,
    val note: String? = null
) {
    companion object {
        /** 从 assets/asr_split_config.json 加载（同源副本，来自 /workspace/shared）。 */
        fun load(context: Context): ASRSplitConfig? = runCatching {
            context.assets.open("asr_split_config.json").bufferedReader().use { reader ->
                Json.decodeFromString<ASRSplitConfig>(reader.readText())
            }
        }.getOrNull()
    }
}

File: ios/DailyPlan/Resources/ASRSplitConfig.swift
// ASRSplitConfig.swift
// 双端共享 ASR 拆分配置解析（规格 §8）。M1 仅定义结构、F2 才消费。
// 重要：禁止在代码中硬编码拆分标点集或阈值，必须从 shared/asr_split_config.json 解析（P0-4）。

import Foundation

struct ASRSplitConfig: Codable {
    let configVersion: Int
    let splitPunctuation: [String]
    let splitPauseThresholdMs: Int
    let includeEnumerationComma: Bool
    let includeNewline: Bool
    // note 字段可选，便于兼容
    let note: String?

    /// 从 App 包内资源加载（拷贝自 /workspace/shared/asr_split_config.json 的同源副本）。
    /// 返回 nil 表示配置文件缺失——调用方应降级为规格默认集，但不得硬编码常量。
    static func loadFromBundle() -> ASRSplitConfig? {
        guard let url = Bundle.main.url(forResource: "asr_split_config", withExtension: "json"),
              let data = try? Data(contentsOf: url) else { return nil }
        return try? JSONDecoder().decode(ASRSplitConfig.self, from: data)
    }
}
